app/src/main/java/net/gsantner/markor/util/CryptWithPassword.java

```diff
@@ -0,0 +1,115 @@
+package net.gsantner.markor.util;
+
+import android.os.Build;
+import android.support.annotation.RequiresApi;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.KeySpec;
+import java.util.Arrays;
+import java.util.Random;
+
+import javax.crypto.Cipher;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.GCMParameterSpec;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * Helper-Class which encrypt a text based on a password.
+ */
+@SuppressWarnings("CharsetObjectCanBeUsed")
+class CryptWithPassword {
+
+    private final int saltLength;
+    private Random random;
+
+    /**
+     * @param random     recommended SecureRandom.getInstanceStrong(), which could be very slow.  A compromise could be SecureRandom.getInstance("SHA1PRNG")
+     * @param saltLength recommended 64
+     */
+    private CryptWithPassword(Random random, int saltLength) {
+        this.random = random;
+        this.saltLength = saltLength;
+    }
+
+    CryptWithPassword() {
+        this(new SecureRandom(), 64);
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.KITKAT)
+    byte[] encrypt(String plainText, char[] password) throws EncryptionFailedException {
+        try {
+            final byte[] salt = getRandomBytes(saltLength);
+            final byte[] nonce = getRandomBytes(32);
+            final SecretKey key = createKeyFromPassword(password, salt);
+            final byte[] cryptedBytes = getCipherGcm(key, Cipher.ENCRYPT_MODE, nonce).doFinal(plainText.getBytes("UTF-8"));
+            final byte[] result = new byte[nonce.length + salt.length + cryptedBytes.length];
+            System.arraycopy(nonce, 0, result, 0, 32);
+            System.arraycopy(salt, 0, result, 32, saltLength);
+            System.arraycopy(cryptedBytes, 0, result, 32 + saltLength, cryptedBytes.length);
+            return result;
+        } catch (Exception e) {
+            throw new EncryptionFailedException("Can't encrypt text.", e);
+        }
+
+    }
+
+
+    @RequiresApi(api = Build.VERSION_CODES.KITKAT)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/manual/gsantner/markor/pullRequest?prid=5222540)